wallet-new/src/Cardano/Wallet/Kernel/DB/HdWallet/Read.hs

```diff
@@ -127,14 +127,29 @@ liftCP f accId =
       zoomHdAccountCurrent $
         asks f
 
+-- | Internal: lift a function on the current checkpoints. The difference with
+--   @liftCP@ is that we pass downstream a function that associates the current
+--   partial and historical checkpoint. This function is only used if the
+--   relevant account is under restoration.
+liftCPwithAssoc :: (a -> a -> a) -> (forall c. IsCheckpoint c => c -> a)
+       -> HdAccountId -> Query' UnknownHdAccount HdWallets a
+liftCPwithAssoc assoc f accId =
```

Comment:
I am not terribly convinced about the concept of "association" here . What you want to convey is that you want, given two choices, to pick one or the other. Perhaps it's worthwhile reusing the existing Haskell vocabulary and go with `Alternative` instead of assoc?

Reply:
Ps. Usually if you want to be strict about camel casing, this should be `liftCpWithAssoc` 😉

Reply:
I find `combine` slightly more descriptive (I don't know why but when I think about "associative" [this](https://en.wikipedia.org/wiki/Associative_property) comes to mind). But if that's just me and my mental model which is wrong, totally happy to stick with `assoc` 😉